lua/core/serial.lua

```diff
@@ -0,0 +1,250 @@
+--- USB serial device interface
+-- @module serial
+
+local serial = {
+    _handlers={},
+}
+
+--- add a new serial device handler
+function serial.handler(args)
+    assert(serial._handlers[args.id] == nil, "duplicate serial handler id: " .. args.id)
+    serial._handlers[args.id] = args
+end
+
+--- send a message to a serial device
+function serial.send(dev, line)
+    _norns.serial_send(dev, line)
+end
+
+_norns.serial = {}
+
+function _norns.serial.match(vendor, model, serial_num, interface_num)
+    for id, handler in pairs(serial._handlers) do
+        if handler.match(vendor, model, serial_num, interface_num) then
+            return id
+        end
+    end
+    return nil
+end
+
+function _norns.serial.configure(handler_id, tio)
+    return serial._handlers[handler_id].configure(tio)
```

Comment:
When I was writing `handler.match` I was thinking it would be useful for it to be able to handle a family of devices that might have different attributes but generally speak the same language. However, `handler.configure` currently doesn't have the same flexibility because all it receives is the termios config. I've been wondering if I need to pass in the same attributes that `handler.match` receives, or give `handler.match` the ability to return some sort of user data that gets passed through to `handler.configure`.

Maybe it's not a big deal though. I think users could work around this limitation by defining multiple serial handlers, but it might be a bit of a headache to wrap that up in a nice interface for scripts.